### ISSUE TITLE
"require-optimization" shouldn't warn on stateless components

### ISSUE THREAD

**Issue by nkt** [yannickcr/eslint-plugin-react#687]
Configuration:
```json
"rules": {
  "react/require-optimization": 2
}
```
Code:
```js
import React from 'react';
import styles from './Spinner.css';

function Spinner() { // Component is not optimized. Please add a shouldComponentUpdate method
  return <div className={styles.spinner} />;
}

export default Spinner;
```

**Comment by nkt** [comment]
Great, thanks!
